automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/NotebookHailSpec.scala

```diff
@@ -0,0 +1,75 @@
+package org.broadinstitute.dsde.workbench.leonardo
+
+import java.io.File
+import java.nio.file.Files
+
+import org.broadinstitute.dsde.workbench.ResourceFile
+import org.broadinstitute.dsde.workbench.auth.AuthToken
+import org.broadinstitute.dsde.workbench.fixture.BillingFixtures
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.scalatest.{BeforeAndAfterAll, FreeSpec}
+
+import scala.concurrent.duration.DurationInt
+import scala.language.postfixOps
+import scala.collection.JavaConverters._
+
+class NotebookHailSpec extends FreeSpec with LeonardoTestUtils with BeforeAndAfterAll with BillingFixtures {
+  /*
+   * This class creates a cluster in a new billing project and runs all tests inside the same cluster.
+   */
+  var gpAllocProject : ClaimedProject = _
+  var billingProject : GoogleProject = _
+  var ronCluster : Cluster = _
+
+  implicit val ronToken: AuthToken = ronAuthToken
+
+  override def beforeAll(): Unit = {
```

Comment:
It seems `beforeAll` / `afterAll` is the same for all 4 test classes: `NotebookHailSpec`, `NotebookInstallSpec`, `NotebookLocalizeFileSpec`, `NotebookPythonKernelSpec`. Could we move some things into a common base class?